rules/Symfony62/Rector/Class_/MessageHandlerInterfaceToAttributeRector.php

```diff
@@ -105,17 +105,18 @@ public function refactor(Node $node): ?Node
     /**
      * @param ServiceDefinition[] $handlers
      */
-    private function checkForServices(Class_ $class, array $handlers): Class_
+    private function checkForServices(Class_ $class, array $handlers): ?Class_
     {
         foreach ($handlers as $handler) {
             if ($this->isName($class, $handler->getClass() ?? $handler->getId())) {
                 $options = $this->messengerHelper->extractOptionsFromServiceDefinition($handler);
                 if (! isset($options['method']) || $options['method'] === '__invoke') {
                     $this->messengerHelper->addAttribute($class, $options);
+                    return $class;
```

Comment:
is multiple `addAttribute()` can happen? if yes, the return can in after foreach, eg, via set `$hasChanged = false` before loop, and set `$hasChanged = true` on this line, then after foreach, verify `if ($hasChanged) { return $class; }`

Reply:
I don't think that you can have multiple `AsMessageHandler` attributes but not 100% sure so I implemented this change just in case

Reply:
> A single handler class can handle multiple messages. For that add the `#AsMessageHandler` attribute to all the handling methods

https://symfony.com/doc/current/messenger.html#handling-multiple-messages

Reply:
@stefantalen the usage of `hasChanged` flag is the safest way for that